src/handlers/admin/menu_handler.py

```diff
@@ -10,187 +11,103 @@
-                    ),
-                )
-            ],
-            [
-                InlineKeyboardButton(
-                    "Изменить сообщение после успешного представления",
-                    callback_data=json.dumps(
-                        {
-                            "chat_id": selected_chat_id,
-                            "action": constants.Actions.set_on_successful_introducion_response,
-                        }
-                    ),
-                )
-            ],
-            [
-                InlineKeyboardButton(
-                    "Изменить сообщение напоминания",
-                    callback_data=json.dumps(
-                        {
-                            "chat_id": selected_chat_id,
-                            "action": constants.Actions.set_notify_message,
-                        }
-                    ),
-                )
-            ],
-            [
-                InlineKeyboardButton(
-                    "Изменить сообщение после кика",
-                    callback_data=json.dumps(
-                        {
-                            "chat_id": selected_chat_id,
-                            "action": constants.Actions.set_on_kick_message,
-                        }
-                    ),
-                )
-            ],
-            [
-                InlineKeyboardButton(
-                    "Получить текущие настройки",
-                    callback_data=json.dumps(
-                        {
-                            "chat_id": selected_chat_id,
-                            "action": constants.Actions.get_current_settings,
-                        }
-                    ),
-                )
-            ],
+def new_button(button_name: str, chat_id, action):
+    return InlineKeyboardButton(button_name, callback_data=json.dumps({"chat_id": chat_id, "action": action,}))
+
+def update_reply_markup(context, keyboard, query):
+    reply_markup = InlineKeyboardMarkup(keyboard)
+    context.bot.edit_message_reply_markup(reply_markup=reply_markup, chat_id = query.message.chat_id, message_id=query.message.message_id)
+
+def handle_start_select_chat(data, query, context, update):
+    with session_scope() as sess:
+        user_id = query.from_user.id
+        users = sess.query(User).filter(User.user_id == user_id)
+        user_chats = [{"title": context.bot.get_chat(x.chat_id).title or x.chat_id, "id": x.chat_id} for x in users]
+
+    if not user_chats:
+        update.message.reply_text("You don't have available chats")
+        return 
+    keyboard = [
+        [new_button(chat["title"], constants.Actions.select_chat, chat["id"])]
+        for chat in user_chats
+        if authorize_user(context.bot, chat["id"], user_id)
+    ]
+    update_reply_markup(context, keyboard, query)
+
+def handle_select_chat(data, query, context):
+    selected_chat_id = data["chat_id"]
+    button_texts = [
+        "Изменить таймаут кика",
+        # ... Other button texts ...
```

Comment:
:heart:

Reply:
I think it would be nice to have an enum for buttons in `constants.py`, where values would correspond to the buttons texts (not applicable for chat selector as text would be the chat name, but that seems to be the only exception), like this:
```python
class Button(Enum):
    SELECT_CHAT = "select_chat_placeholder"
    SET_KICK_SETTINGS = "Настройки кика"
    SET_KICK_TIMEOUT = "Изменить таймаут кика"
    ...
```

And also have a `callback_map` dict, which would link the buttons to the corresponding callbacks, for example:
```
callback_map = {
    Button.SELECT_CHAT: handle_select_chat
    Button.SET_KICK_SETTINGS: handle_set_kick_settings  # will return a menu with SET_KICK_TIMEOUT button etc
    Button.SET_KICK_TIMEOUT: message_handler
}
```

That way we will effectively replace the `Actions` enum, but this is better because
1) the enum values will start to have meaning (button text, which can be later changed into button_text_id for localization);
2) the `Actions` enum is just a list of identifiers which are later mapped to the callbacks functions through `if`s, which IMO is more wordy and confusing.

Reply:
Also it is better to rename `handle_action`, `handle_select_chat` from `handle_` to `callback_`, that way we'll have a semantic separation between handlers (for bot /commands) and callbacks (triggered by menu buttons).